### PR TITLE
chore(invoice): Clean up layout

### PR DESCRIPTION
**Pull request by damassi**
The type of this PR is: **Chore**

### Description

This adds a bit of polish and visual simplification to the ui, and matches some other patterns seen elsewhere: 

Screenshot 2024-10-25 at 11 43 53 AM

Screenshot 2024-10-25 at 11 51 27 AM

cc @artsy/diamond-devs

### CODE REVIEW

**Review by mzikherman — DISMISSED** [review]
Nice! I like it.

File: src/Apps/Invoice/Routes/__tests__/InvoiceDetailRoute.jest.tsx [hunk 1]
```diff
@@ -68,10 +68,7 @@ describe("InvoiceDetailRoute", () => {
   it("renders invoice details correctly", () => {
     expect(
       screen.getByTestId("invoice-name").textContent
-    ).toMatchInlineSnapshot(`"Name: John Doe"`)
-    expect(
-      screen.getByTestId("invoice-email").textContent
-    ).toMatchInlineSnapshot(`"Email: john@example.com"`)
+    ).toMatchInlineSnapshot(`"John Doe <john@example.com>"`)
```
Comment (mzikherman) [inline src/Apps/Invoice/Routes/__tests__/InvoiceDetailRoute.jest.tsx:71]: This actually reminded me that my main question/'issue' around this super simple app was the testing. I added these just as a basic 'info properly rendered', however `getByText` was proving problematic due to use of `<em>` and the text spanning multiple elements.

I even went down a little rabbit hole of writing a custom matcher/regex, before just doing a snapshot (which I don't love).

Do you recommend against snapshots (I feel like they're easily overlooked plus are perfectly brittle), or was it maybe a smell that I was finding it hard to right vanilla text matchers (I shouldn't be using `<em>` like that and so forth)?

Comment (damassi) [reply src/Apps/Invoice/Routes/__tests__/InvoiceDetailRoute.jest.tsx:71]: Yah, 100% avoid snapshots. 

```
expect(screen.getByText("Foo")).toBeInTheDocument()
```
just keep it simple

Comment (damassi) [reply src/Apps/Invoice/Routes/__tests__/InvoiceDetailRoute.jest.tsx:71]: But to answer the q, to fix that problem, you do a regex:

```
screen.getByText(/foo bar baz/)
```
that'll cross over line breaks, etc.

Comment (damassi) [reply src/Apps/Invoice/Routes/__tests__/InvoiceDetailRoute.jest.tsx:71]: (Removed snapshots in here: https://github.com/artsy/force/pull/14733)

**Pull request merged**